test/infrastructure/docker/docker/types/node.go

```diff
@@ -176,45 +176,95 @@ func (c *containerCmd) RunLoggingOutputOnFail(ctx context.Context) ([]string, er
 }
 
 func (c *containerCmd) Run(ctx context.Context) error {
-	args := []string{
-		"exec",
+	cli, err := dockerClient.NewClientWithOpts(dockerClient.FromEnv, dockerClient.WithAPIVersionNegotiation())
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	execConfig := &dockerTypes.ExecConfig{
 		// run with privileges so we can remount etc..
 		// this might not make sense in the most general sense, but it is
 		// important to many kind commands
-		"--privileged",
-	}
-	if c.stdin != nil {
-		args = append(args,
-			"-i", // interactive so we can supply input
-		)
+		Privileged:   true,
+		Cmd:          append([]string{c.command}, c.args...), // with the command specified
+		AttachStdout: true,
+		AttachStderr: true,
+		AttachStdin:  (c.stdin != nil), // interactive so we can supply input
 	}
+
 	// set env
-	for _, env := range c.env {
-		args = append(args, "-e", env)
-	}
-	// specify the container and command, after this everything will be
-	// args the the command in the container rather than to docker
-	args = append(
-		args,
-		c.nameOrID, // ... against the container
-		c.command,  // with the command specified
-	)
-	args = append(
-		args,
-		// finally, with the caller args
-		c.args...,
-	)
-	cmd := exec.CommandContext(ctx, "docker", args...)
-	if c.stdin != nil {
-		cmd.SetStdin(c.stdin)
+	execConfig.Env = append(execConfig.Env, c.env...)
+
+	response, err := cli.ContainerExecCreate(ctx, c.nameOrID, *execConfig)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	execID := response.ID
+	if execID == "" {
+		return errors.New("exec ID empty")
+	}
+
+	fmt.Printf("++++\nAttaching to exec: %s\n%+v\n++++\n", execID, execConfig)
+	resp, err := cli.ContainerExecAttach(ctx, execID, dockerTypes.ExecStartCheck{})
+	if err != nil {
+		return errors.WithStack(err)
 	}
+	defer resp.Close()
+
+	var (
+		out, stderr io.Writer
+	)
+
 	if c.stderr != nil {
-		cmd.SetStderr(c.stderr)
+		stderr = c.stderr
+	} else {
+		stderr = os.Stderr
```

Comment:
I'm not sure attaching to os.Stderr make sense inside a controller... What are you trying to achieve here?

Reply:
I was trying to default to the behavior when calling the CLI. I think I could just skip over reading output below if c.stderr and c.stdout are not set and just continue on to the `ContainerExecInspect` call to get the StatusCode.